src/guidellm/executor/profile_generator.py

```diff
@@ -61,35 +67,41 @@ def next_profile(
         pass
 
 
-@ProfileGenerator.register_generator(ProfileGenerationModes.SINGLE)
-class SingleProfileGenerator(ProfileGenerator):
-    def __init__(self, rate: float, rate_type: str, **kwargs):
-        super().__init__(ProfileGenerationModes.SINGLE)
-        self._rate = rate
-        self._rate_type = rate_type
+@ProfileGenerator.register_generator(ProfileGenerationModes.FIXED)
+class FixedRateProfileGenerator(ProfileGenerator):
+    def __init__(self, rate_type: str, rate: Optional[List[float]] = None, **kwargs):
+        super().__init__(ProfileGenerationModes.FIXED)
+        if rate_type == "synchronous" and rate and len(rate) > 0:
+            raise ValueError("custom rates are not supported in synchronous mode")
+        self._rates = rate
+        self._rate_index = 0
         self._generated = False
+        self._rate_type = rate_type
 
     def next_profile(
         self, current_report: TextGenerationBenchmarkReport
     ) -> Optional[Profile]:
-        if self._generated:
-            return None
-
-        self._generated = True
-
-        if self._rate_type == "constant":
-            return Profile(
-                load_gen_mode=LoadGenerationModes.CONSTANT, load_gen_rate=self._rate
-            )
-
         if self._rate_type == "synchronous":
+            if self._generated:
+                return None
+        
+            self._generated = True
+
             return Profile(
                 load_gen_mode=LoadGenerationModes.SYNCHRONOUS, load_gen_rate=None
             )
```

Comment:
Part of me feels like this class is now a little encumbered handling both the synchronous and async profile generator rate types since they don't benefit from being in the same class here.

I have half a mind to split this out into a SynchronousProfileGenerator but I don't have a good sense of that is actually a good idea or only seems more convenient now. Is the synchronous rate type totally distinct like it seems? If so I think that'd be a cleaner way to handle this.